ukfom/ukf.hpp

```diff
@@ -187,32 +187,30 @@ class ukf {
 	{
 		assert(X.size() == 2 * n + 1);
 
-		lapack::cholesky<n> L(sigma);
+        Eigen::LLT<cov> lltOfSigma(sigma); // compute the Cholesky decomposition of A
+        cov L = lltOfSigma.matrixL(); // retrieve factor L  in the decomposition
 
-		if (!L.isSPD())
-		{
-			std::cerr << std::endl << "sigma is not SPD:" << std::endl
-					  << sigma << std::endl
-					  << "---" << std::endl;
-			Eigen::EigenSolver<cov> eig(sigma);
-			std::cerr << "eigen values: " << eig.eigenvalues().transpose() << std::endl;
-		}
-		
-		assert(L.isSPD());
+        if (lltOfSigma.info() != Eigen::ComputationInfo::Success)
+        {
+            std::cerr << std::endl << "sigma is not SPD:" << std::endl
+            << sigma << std::endl
+            << "---" << std::endl;
+            Eigen::EigenSolver<cov> eig(sigma);
```

Comment:
This can actually be replaced by the more stable and faster
  `Eigen::SelfAdjointEigenSolver<cov> eig(sigma, Eigen::EigenvaluesOnly);`